### PR TITLE
Fixing OpenSSL Deprecation issues

### PR DESCRIPTION
**Pull request by adarshan-intel**
Fixing OpenSSL Deprecation issues for the functions 
1. test_fdo_cryptoECDSASign
2. test_crypto_support_Private_key
image

### CODE REVIEW

File: .gitignore
Comment (shrikant1407) [inline .gitignore:1]: remove this file

File: build.sh [hunk 1]
```diff
@@ -87,12 +87,12 @@ build_bin x86_ecdsa256_gcm_bin -DAES_MODE=gcm -DDA=ecdsa256
...
-build_bin tpm_ecdsa384_gcm_bin -DAES_MODE=gcm -DDA=tpm20_ecdsa384
-build_bin tpm_ecdsa384_ccm_bin -DAES_MODE=ccm -DDA=tpm20_ecdsa384
-build_bin cse_ecdsa384_gcm_bin -DAES_MODE=gcm -DDA=cse_ecdsa384 -DCSE_CLEAR=true
-build_bin cse_ecdsa384_ccm_bin -DAES_MODE=ccm -DDA=cse_ecdsa384 -DCSE_CLEAR=true
+# build_bin tpm_ecdsa256_gcm_bin -DAES_MODE=gcm -DDA=tpm20_ecdsa256
+# build_bin tpm_ecdsa256_ccm_bin -DAES_MODE=ccm -DDA=tpm20_ecdsa256
+# build_bin tpm_ecdsa384_gcm_bin -DAES_MODE=gcm -DDA=tpm20_ecdsa384
+# build_bin tpm_ecdsa384_ccm_bin -DAES_MODE=ccm -DDA=tpm20_ecdsa384
+# build_bin cse_ecdsa384_gcm_bin -DAES_MODE=gcm -DDA=cse_ecdsa384 -DCSE_CLEAR=true
+# build_bin cse_ecdsa384_ccm_bin -DAES_MODE=ccm -DDA=cse_ecdsa384 -DCSE_CLEAR=true
```
Comment (shrikant1407) [inline build.sh:95]: uncomment these lines

File: tests/unit/test_ECDSASignRoutines.c [hunk 2]
```diff
@@ -238,16 +245,72 @@ TEST_CASE("crypto_hal_ecdsa_sign", "[ECDSARoutines][fdo]")
...
+	}
+
+	int group_nid = OBJ_sn2nid(group_name);
+	const EC_GROUP *group = EC_GROUP_new_by_curve_name(group_nid);
+
+	if (group == NULL) {
+		LOG(LOG_ERROR, "Failed to get the group name fo EC EVP key\n");
+		exit(1);
+	}
+	//  const EC_GROUP *group = EC_KEY_get0_group(avalidkey);
```
Comment (shrikant1407) [inline tests/unit/test_ECDSASignRoutines.c:296]: remove this code if not required

Comment (adarshan-intel) [reply tests/unit/test_ECDSASignRoutines.c:296]: Only the comments part right?

**Review by DukeDavis12 — DISMISSED** [review]

**Comment by shrikant1407** [comment]
Uncomment these tests from CMakeFile and add it to this PR also remove the commented code

**Review by KiranSukhavasi — APPROVED** [review]

**Comment by adarshan-intel** [comment]
> Uncomment these tests from CMakeFile and add it to this PR also remove the commented code

Made a commit, please check now

**Review by shrikant1407 — DISMISSED** [review]

**Comment by adarshan-intel** [comment]
@shrikant1407 @KiranSukhavasi @DukeDavis12 Please review as I had update the branch and previous review became stale.

**Comment by adarshan-intel** [comment]
@shrikant1407 @KiranSukhavasi @DukeDavis12 Please review again.

**Review by KiranSukhavasi — APPROVED** [review]

**Review by DukeDavis12 — APPROVED** [review]

**Pull request merged**